### PR TITLE
#163388215 Users will Fetch Single Article, Update and Delete the article

### PR DESCRIPTION
**Pull request by calebrotich**
#### What does this PR do?
This PR will enable a user to view a single article, update the article if it is theirs and delete the article

#### Description of Task to be completed?
- User to fetch a single article
- User to update an article
- User to delete the article
- Improves the navigation bar for the app
- Introduces loaders to the application

#### How should this be manually tested?
(_Note: Screenshots below indicate valid urls(add `s` to `article`) and slug which can be used for test_)
__Fetch Single Article__
Follow the address `/articles/:slug` to view the article with the specified slug

__Update Article__
From the above address, `/articles/:slug` click the edit icon on the right of the article, the icon appears only if you own the article

__Delete Article__
From the same address, `/article/:slug`, find the delete icon at the right of the page

#### What are the relevant pivotal tracker stories?
[#163388215](https://www.pivotaltracker.com/story/show/163388215)

#### Screenshots (if appropriate):
_1. Article Edit Page_
image

_2. Single Article page with update and delete controls_
image

#### Reviewers:
@TooColline
@calebrotich10
@unah254
@bryan-munene
@hoslack

### CODE REVIEW

**Review by unah254 — APPROVED** [review]
LGTM

File: src/App.jsx [hunk 1]
```diff
@@ -28,6 +30,8 @@ class App extends Component {
             <Route exact path="/verify/:token" component={verify} />
             <Route path="/getprofile" component={Profile} />
             <Route path="/edit-profile" component={EditProfile} />
+            <Route exact path="/article/:slug" component={SingleArticle} />
```
Comment (ezrqnkemboi) [inline src/App.jsx]: Could have been better to make use of url `/articles/slug`

File: src/App.jsx [hunk 2]
```diff
@@ -28,6 +30,8 @@ class App extends Component {
...
+            <Route exact path="/article/edit/:slug" component={EditArticle} />
```
Comment (ezrqnkemboi) [inline src/App.jsx]: See comment above.

**Review by TooColline — APPROVED** [review]
This is good work

**Review by hoslack — APPROVED** [review]
Great job!

**Pull request merged**